src/Microsoft.ML.Dnn/LearningRateScheduler.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Transforms
+{
+    /// <summary>
+    /// A class that contains the current train state to use for learning rate scheduling.
+    /// </summary>
+    internal class TrainState
+    {
+        internal int CurrentBatchIndex;
+        internal int CurrentEpoch;
+        internal int BatchSize;
+        internal int BatchesPerEpoch;
+    }
+
+    /// <summary>
+    /// This interface defines a learning rate scheduler.
+    /// </summary>
+    public abstract class LearningRateScheduler
+    {
```

Comment:
Make an `internal` constructor on `LearningRateScheduler`, so no one outside of this assembly can derive from it.